cmd/derper/bootstrap_dns.go

```diff
@@ -33,8 +34,15 @@ var (
 	publishedDNSMisses   = expvar.NewInt("counter_bootstrap_dns_published_misses")
 	unpublishedDNSHits   = expvar.NewInt("counter_bootstrap_dns_unpublished_hits")
 	unpublishedDNSMisses = expvar.NewInt("counter_bootstrap_dns_unpublished_misses")
+	bootstrapLookupMap   = make(map[string]bool)
```

Comment:
I believe this is racy, since the metrics scrape is handled in a different goroutine from where these are added. 

Maybe use an expvar.Map just for its baked in synchronization?

Reply:
It would. I was imagining you wouldn't publish it directly, but just use it as a repository for the data to publish via the Func.

But that would be a kluge, and your solution is better. 😄